File: JavaScript/boj/Silver/1181.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const [N, ...input] = fs.readFileSync(filePath).toString().trim().split('\n');

const words = [...new Set(input)];

words.sort((a, b) => {
  if (a.length === b.length) return a.localeCompare(b);
  return a.length - b.length;
});

console.log(words.join('\n'));

File: Python/boj/Silver/1181.py
import sys

if __name__ == "__main__":
    try:
        sys.stdin = open("input.txt", "r")
    except FileNotFoundError:
        pass

    N = int(input())
    words = list(set([input() for _ in range(N)]))

    words.sort(key=lambda x: (len(x), x))
    print("\n".join(words))
